scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java

```diff
@@ -749,19 +781,59 @@ private void dispatchJobStateUpdated(String owner, NotificationData<JobInfo> not
                 }
             }
             for (ListeningUser listeningUserId : identifications.values()) {
-                //if this user has a listener
+                // if this user has a listener
                 if (listeningUserId.isListening()) {
                     UserIdentificationImpl userId = listeningUserId.getUser();
-                    //if there is no specified event OR if the specified event is allowed
+                    // if there is no specified event OR if the specified event
+                    // is allowed
                     if ((userId.getUserEvents() == null) ||
                         userId.getUserEvents().contains(notification.getEventType())) {
-                        //if this userId have the myEventOnly=false or (myEventOnly=true and it is its event)
+                        // if this userId have the myEventOnly=false or
+                        // (myEventOnly=true and it is its event)
                         if (!userId.isMyEventsOnly() ||
                             (userId.isMyEventsOnly() && userId.getUsername().equals(owner))) {
-                            listeningUserId.getListener().addEvent(eventMethods.get("jobStateUpdatedEvent"),
-                                    notification);
+                            listeningUserId.getListener()
+                                    .addEvent(eventMethods.get(JOB_STATE_UPDATED_EVENT_METHOD), notification);
+                        }
+                    }
+                }
+            }
+            clearListeners();
+        } catch (SecurityException e) {
+            logger.error("", e);
+        }
+    }
+
+    /**
+     * Dispatch the job state updated event
+     * 
+     * @param owner
+     *            the owner of this job
+     * @param notification
+     *            the data to send to every client
+     */
+    private void dispatchJobUpdatedFullData(JobState job) {
+        try {
+            if (logger.isDebugEnabled()) {
+                jlogger.debug(job.getJobInfo().getJobId(), " event [" + SchedulerEvent.JOB_UPDATED + "]");
+            }
+            for (ListeningUser listeningUserId : identifications.values()) {
+                // if this user has a listener
+                if (listeningUserId.isListening()) {
+                    UserIdentificationImpl userId = listeningUserId.getUser();
+                    // if there is no specified event OR if the specified
+                    // event is allowed
+                    if ((userId.getUserEvents() == null) ||
+                        userId.getUserEvents().contains(SchedulerEvent.JOB_UPDATED)) {
+                        // if this userId have the myEventOnly=false or
+                        // (myEventOnly=true and it is its event)
+                        if (!userId.isMyEventsOnly() ||
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1066)